src/services/navigation/index.js

```diff
@@ -68,6 +68,7 @@ export default class Navigation {
     element: Element,
     formComponents: ComponentRegistry,
     opts: BehaviorOptions,
+    registerPreload: ?(id: number, element: Element) => void,
```

Comment:
Does this need a question mark before AND after the colon? I've seen it three different ways in the Javascript

Reply:
Before the colon means the key (`registerPreload`) is optional, it doesn't need to be included in the object. But if the key is included, the value must be a callback:
```js
// registerPreload?: (id: number, element: Element) => void

{ } // valid
{ registerPreload: (id, element) => {} } // valid
{ registerPreload: undefined } // invalid
```

After the colon means the key must always be present, but the value can be undefined:
```js
// registerPreload: ?(id: number, element: Element) => void

{ } // invalid
{ registerPreload: (id, element) => {} } // valid
{ registerPreload: undefined } // valid
```